packages/ppbgdi/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -329,24 +378,202 @@ processors:
+  - remove:
+      field: _tmp.sc_range_end
+      ignore_missing: true
+
 
+  ### Cloudfront standard-log fields ###########################################
 
-### Additional fields ##############################################################################
+  # date
+  # time
+  - script:
+      description: timestamp composed of date and time. Script processor performs better than equivalent set processor.
+      lang: painless
+      source: >-
+        ctx._tmp.timestamp = ctx._tmp.date + 'T' + ctx._tmp.time;
+      if: ctx._tmp?.date != null && ctx._tmp?.time != null
+  - remove:
+      field: _tmp.date
+      ignore_missing: true
+  - remove:
+      field: _tmp.time
+      ignore_missing: true
+  # timestamp (Available in realtime logs instead of date and time fields)
+  - date:
+      field: _tmp.timestamp
+      target_field: '@timestamp'
+      ignore_failure: true
+      formats:
+        - ISO8601
+        - yyyy-MM-dd HH:mm:ss
+      if: ctx._tmp.timestamp != null
+  - remove:
+      field: _tmp.timestamp
+      ignore_missing: true
+
+
+  ### Cloudfront realtime-log fields ###########################################
+
+  # timestamp
+  - date:
+      field: _tmp.timestamp
+      target_field: '@timestamp'
+      formats:
+        - UNIX
+        - ISO8601
+        - yyyy-MM-dd HH:mm:ss
+      if: ctx._tmp?.timestamp != null
+  - remove:
+      field: _tmp.timestamp
+      ignore_missing: true
+  # c_ip_version
+  - rename:
+      field: _tmp.c_ip_version
+      target_field: network.type
+      ignore_missing: true
+  # c_country
+  - rename:
+      field: _tmp.c_country
+      target_field: aws.cloudfront.client.country
+      ignore_missing: true
+  # cs_accept_encoding
+  - rename:
+      field: _tmp.cs_accept_encoding
+      target_field: http.request.headers.Accept-Encoding
+      ignore_missing: true
+  # cs_accept
+  - rename:
+      field: _tmp.cs_accept
+      target_field: http.request.headers.Accept
+      ignore_missing: true
+  # cache_behavior_path_pattern
+  - rename:
+      description: Cloudfront realtime-logs field
+      field: _tmp.cache_behavior_path_pattern
+      target_field: aws.cloudfront.cache_behavior.path_pattern
+      ignore_missing: true
+  # cs_headers
+  - urldecode:
+      field: _tmp.cs_headers
+      ignore_missing: true
+  - kv:
+      field: _tmp.cs_headers
+      description: Parse request headers string into individual header fields.
+      target_field: http.request.headers
+      field_split: '\n'
+      value_split: ':'
+      ignore_missing: true
+  - remove:
+      field: _tmp.cs_headers
+      ignore_missing: true
+  - split:
+      field: http.request.headers.accept
+      separator: ','
+      ignore_missing: true
+  - split:
+      field: http.request.headers.accept-encodings
+      separator: ','
+      ignore_missing: true
+  - split:
+      field: http.request.headers.accept-languages
+      separator: ','
+      ignore_missing: true
```

Comment:
one thing I noted is that (at least in some cases) both uppercase and lowercase fields of the header names exist, see e.g. https://kibana-dev.geobax.tech/app/discover#/doc/4f24c28d-48c7-498e-9d43-e634b76a12dc/.ds-logs-ppbgdi.cloudfront_logs-dev-2024.10.06-000021?id=816636c936
dunno if this should be fixed here as well to only have lowercase header names?